File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/ordenes_producto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

[PrimaryKey("id_orden_compra", "id_producto")]
public partial class ordenes_producto
{
    [Key]
    public int id_orden_compra { get; set; }

    [Key]
    public int id_producto { get; set; }

    [Required(ErrorMessage = "La cantidad es obligatoria.")]
    [Range(1, int.MaxValue, ErrorMessage = "La cantidad debe ser mayor que cero.")]
    public int cantidad { get; set; }

    [ForeignKey("id_orden_compra")]
    [InverseProperty("ordenes_productos")]
    public virtual ordenes_de_compra id_orden_compraNavigation { get; set; } = null!;

    [ForeignKey("id_producto")]
    [InverseProperty("ordenes_productos")]
    public virtual producto id_productoNavigation { get; set; } = null!;
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/ordenesProducto.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

// Esquema del modelo OrdenesProducto
const ordenesProductoSchema = new Schema({
    id_orden_compra: {
        type: Schema.Types.ObjectId,
        required: true,
        ref: 'ordenesDeCompra' // Asegúrate de tener el modelo "OrdenesDeCompra" definido
    },
    id_producto: {
        type: Schema.Types.ObjectId,
        required: true,
        ref: 'producto' // Asegúrate de tener el modelo "Producto" definido
    },
    cantidad: {
        type: Number,
        required: [true, 'La cantidad es obligatoria.'],
        min: [1, 'La cantidad debe ser mayor que cero.']
    }
});

// Crea el modelo OrdenesProducto
const OrdenesProducto = mongoose.model('OrdenesProducto', ordenesProductoSchema);

module.exports = OrdenesProducto;
